app/cdap/components/CreateAndEditRecipeForm/index.tsx

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { FormEvent, useEffect, useState, useRef } from 'react';
+import { FormControl } from '@material-ui/core';
+import T from 'i18n-react';
+import {
+  ICreateAndEditRecipeFormProps,
+  IRecipeData,
+} from 'components/CreateAndEditRecipeForm/types';
+import {
+  FormFieldWrapper,
+  NormalLabel,
+  StyledForm,
+  StyledRecipeNameTextField,
+  StyledDescriptionTextArea,
+  StyledCancelButton,
+  StyledSaveButton,
+  StyledFormButtonWrapper,
+  getLabelStyle,
+} from 'components/CreateAndEditRecipeForm/styledComponents';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import MyDataPrepApi from 'api/dataprep';
+import { debounce } from 'lodash';
+
+export default function({
+  recipeData,
+  setIsCreateAndEditRecipeFormOpen,
+  setSnackbar,
```

Comment:
do we need the dependency on parent's snackbar
can't we use the snack bar directly from this component?

plz check and let me know with proper explanation
what went behind this thought and why we should or shouldn't change this

Reply:
for edit we are using panel so whenever we get API success panel will be closed automatically so to show snackbar even after closing the panel we need dependency on parent's snackbar